influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -576,7 +661,18 @@ pub(crate) mod tests {
 
         let catalog = Catalog::new();
 
-        let lp = "cpu,tag1=cupcakes bar=1 10\nmem,tag2=turtles bar=3 15\nmem,tag2=snakes bar=2 20";
+        // When we persist the data all of these duplicates should be removed
+        let lp = "cpu,tag1=cupcakes bar=1 10\n\
+                  cpu,tag1=cupcakes bar=1 10\n\
+                  cpu,tag1=cupcakes bar=1 10\n\
```

Comment:
Can you update the test to have a line `cpu,tag1=something bar=5 10` and then verify that the cpu table has the two rows? I just want to make absolutely certain that it's not deduplicating data with the same timestamp, but different tag values.

Reply:
It dedupes by tag value not timestamp alone. Test updated here -> https://github.com/influxdata/influxdb/pull/24870/commits/a1a8dfa52f44a36fff5d0070d776234682d70938